src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -196,7 +196,7 @@ u16 EnGo_GetTextID(PlayState* play, Actor* thisx) {
 }
 
 s16 EnGo_SetFlagsGetStates(PlayState* play, Actor* thisx) {
-    s16 unkState = 1;
+    s16 unkState = NPC_TALK_STATE_TALKING;
```

Comment:
unkState -> talkState
(while you're at it)

also this function is a getTalkState basically but it has side effects like setting flags, this seems like a broad pattern, we may need to not name these like simple getters?

Reply:
Do you mean the specific functions in actors like `SetFlagsGetStates` here or the callback name in `Npc_UpdateTalking` / `NpcGetTalkStateFunc`?

Indeed almost all actors have side effects (flags or sound effects or animations) in the callback function; that's somewhat understandable because that's where the actor can notice changes in the dialog. One possibility could be to rename `GetTalkState` -> `AdvanceTalkState` or something, that could better convey that the function may also change the actor's own state in addition to returning the new talk state value.

Reply:
> Do you mean the specific functions in actors like SetFlagsGetStates here or the callback name in Npc_UpdateTalking / NpcGetTalkStateFunc?

both (the callback functions in the actors should be named after the callback type name if possible)

> AdvanceTalkState or something, that could better convey that the function may also change the actor's own state in addition to returning the new talk state value.

Yeah, UpdateTalkState would be the typical way to name it but idk maybe AdvanceTalkState would be better

I'd wait for more opinions before switching

Reply:
Personally I'm fine with `UpdateTalkState`. Also just to throw out another idea `HandleTalkState`?

Reply:
I would settle on `UpdateTalkState` (what a predictable outcome)